### ISSUE TITLE
We can see correct output in log but cannot coerce it into text output

### ISSUE THREAD

**Issue by bobthacker** [arborworkflows/ArborWebApps#130]
Try out the sponge > parafitAnalysisScratch
This just runs an example from ape that creates a parafit object for output.
The output is fine in the log, but how to write it into an Arbor output file?

**Comment by curtislisle** [comment]
Isn’t that what this analysis does?  the rr output (a data frame) is correctly listed as a data frame output, which is a type of table.  Therefore, it is displayable as a table from the visualization tab and downloadable as a (slightly strange, I’ll give you that) CSV file.  I am missing the point, I think, how is this malfunctioning? 




x
""
Test of host-parasite coevolution 
 
"Global test:  ParaFitGlobal = 0.01389872 , p-value = 0.01 ( 99 permutations) "
 
There are 17 host-parasite links in matrix HP 
 
Test of individual host-parasite links ( 99 permutations) 
 
      Host Parasite      F1.stat p.F1    F2.stat p.F2
" [1,]    1        2 0.0009312199 0.02 0.08323295 0.01"
" [2,]    1        8 0.0011611181 0.09 0.10378138 0.02"
" [3,]    2        1 0.0010501927 0.03 0.09386681 0.01"
" [4,]    2        9 0.0006711532 0.23 0.05998804 0.03”

etc.

**Comment by lukejharmon** [comment]
We'll sort of! But this is a hack, ideally we would display the lines of text.

Sent from my iPhone

On May 8, 2015, at 6:16 PM, Curtis Lisle <notifications@github.com<mailto:notifications@github.com>> wrote:

Isn't that what this analysis does? the rr output (a data frame) is correctly listed as a data frame output, which is a type of table. Therefore, it is displayable as a table from the visualization tab and downloadable as a (slightly strange, I'll give you that) CSV file. I am missing the point, I think, how is this malfunctioning?




x
""
Test of host-parasite coevolution

"Global test: ParaFitGlobal = 0.01389872 , p-value = 0.01 ( 99 permutations) "

There are 17 host-parasite links in matrix HP

Test of individual host-parasite links ( 99 permutations)

Host Parasite F1.stat p.F1 F2.stat p.F2
" [1,] 1 2 0.0009312199 0.02 0.08323295 0.01"
" [2,] 1 8 0.0011611181 0.09 0.10378138 0.02"
" [3,] 2 1 0.0010501927 0.03 0.09386681 0.01"
" [4,] 2 9 0.0006711532 0.23 0.05998804 0.03"

etc.




On May 8, 2015, at 4:19 PM, bobthacker <notifications@github.com<mailto:notifications@github.com>> wrote:

> Try out the sponge > parafitAnalysisScratch
> This just runs an example from ape that creates a parafit object for output.
> The output is fine in the log, but how to write it into an Arbor output file?
>
> -
> Reply to this email directly or view it on GitHub.
>


-

**Comment by bobthacker** [comment]
Curt, you are looking at it after we had a two hour session with Luke to get to this point.

It looks MUCH nicer just as R output.

We tried several ways to coerce the parafit object into a simple display of text, with all but the dataframe being total fails. String should work but does not.

The current output looks awful compared to how it would look if we could just capture.output, sink, or print the actual text output to a text file.

This issue applies to MANY R functions -- it would be most useful to write text to a text file and then display the text file. The console log can do this even, so it seems like it should not be difficult to have this type of output in arbor.

**Comment by curtislisle** [comment]
I agree that analyses need to be able to consume and generate strings.  Y'all shouldn't have to pack things into dataframes.  We will keep this issue open so it can be addressed by adding string I/O

**Comment by jeffbaumes** [comment]
String output should be working - there was an issue with visualization that has now been resolved. String input as a dataset instead of manually entered text seems to be the missing piece for full string I/O support. We can put this one on our list.
